src/ctelnet.cpp

```diff
@@ -2730,6 +2771,86 @@ void cTelnet::processTelnetCommand(const std::string& telnetCommand)
             break;
         }
 
+        case OPT_COMPRESS4: {
+            if (telnetCommand.length() >= 5 && telnetCommand[3] == MCCP4_BEGIN_ENCODING) {
+                // Extract encoding type from BEGIN_ENCODING suboption
+                std::string encodingType;
+                for (size_t i = 4; i < telnetCommand.length() - 2; ++i) {
+                    if (telnetCommand[i] == TN_IAC) {
+                        break; // End of encoding string
+                    }
+                    encodingType += telnetCommand[i];
+                }
+
+                if (encodingType == "zstd") {
+                    mMCCP4_encoding = 1; // zstd encoding
+                    qDebug() << "MCCP4: Server selected zstd encoding";
```

Comment:
Is "encoding"/"encodingType" the right word for this, surely it actually is "compressionMethod"/"compressionType"...?